interop-testing/src/main/java/io/grpc/testing/integration/TrafficControlProxy.java

```diff
@@ -0,0 +1,245 @@
+package io.grpc.testing.integration;
+
+import io.netty.channel.socket.nio.NioSocketChannel;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+// import java.lang.annotation.Native;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketException;
+import java.net.SocketImpl;
+import java.util.ArrayList;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.logging.Logger;
+
+public class TrafficControlProxy {
+
+  private static int DEFAULT_BAND = 1 * 1024 * 1024; // Bps
+  private static int DEFAULT_DELAY_MS = 100; // ms;
+  private static Logger logger = Logger.getLogger(TrafficControlProxy.class.getName());
+
+  private String loopback = "127.127.127.127";
+  // TODO: make ports arguments
+  private int proxyPort = 5050;
+  private int serverPort = 5001;
+  private int qLength;
+  private int serverQSize;
+  private int clientQSize;
+  private int chunkSize;
+  private boolean shutDown;
```

Comment:
This is accessed from two different threads, so it needs to be properly synchronized.  You can do this by making it volatile, or adding it in a synchronized block.